camp/camp-brooklyn/src/main/java/org/apache/brooklyn/camp/brooklyn/spi/creation/BrooklynComponentTemplateResolver.java

```diff
@@ -509,13 +509,11 @@ public EntitySpecSupplier(EntitySpecConfiguration flag) {
                 this.flag = flag;
             }
             @Override public EntitySpec<?> get() {
-                EntitySpecConfiguration specConfig = (EntitySpecConfiguration) flag;
                 // TODO: This should called from BrooklynAssemblyTemplateInstantiator.configureEntityConfig
                 // And have transformSpecialFlags(Object flag, ManagementContext mgmt) drill into the Object flag if it's a map or iterable?
                 @SuppressWarnings("unchecked")
-                Map<String, Object> resolvedConfig = (Map<String, Object>)transformSpecialFlags(specConfig.getSpecConfiguration());
-                specConfig.setSpecConfiguration(resolvedConfig);
```

Comment:
Good fix as well. Now I think `setSpecConfiguration` method should be also removed.
BROOKLYN-567